packages/astro/src/internal/index.ts

```diff
@@ -184,7 +184,17 @@ export async function renderComponent(result: any, displayName: string, Componen
     }
   }
 
-  ({ html } = await renderer.ssr.renderToStaticMarkup(Component, props, children));
+  if(renderer === null) {
+    if(typeof Component === 'string') {
+      const propsArray = Object.entries(props);
+      html = await renderAstroComponent(await render`<${Component}${ propsArray.length ? ' ' + propsArray.map(([k,v]) => `"${k}"="${v}"`).join(' ') : ''}>${children}</${Component}>`);
```

Comment:
We have a util for rendering props to a string! `destructureProps` in `astro/internal` I think